standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1201,22 +1199,86 @@ private TableValidWriteIds getValidWriteIdsForTable(Statement stmt, String fullT
     }
   }
 
+  private List<Long> getValidTxnIds(AllocateTableWriteIdsRequest rqst, Statement stmt, ResultSet rs)
+          throws MetaException, SQLException {
+    List<Long> txnIds = new ArrayList<>();
+    String dbName = rqst.getDbName().toLowerCase();
+    String tblName = rqst.getTableName().toLowerCase();
+    Long validWriteId;
+    Long largestWriteId, smallestWriteId;
+    List<TxnToWriteId> txnToWriteIds = rqst.getTxnToWriteIdList();
+    assert (txnToWriteIds != null);
+
+    // Get the next write id for the table and check if its in consistent with the input txntowriteids.
+    String s = sqlGenerator.addForUpdateClause(
+            "select nwi_next from NEXT_WRITE_ID where nwi_database = " + quoteString(dbName)
```

Comment:
Next_write_id is queried twice in this allocate write id flow. One here and other by caller after this call.
Add a comment, that this query is run under read-committed connection but still it is guaranteed that repeated query on next_write_id within this connection is consistent as this is repl flow and we  assume that only repl task would allocate write id. 
Also, check if we can refactor and avoid this repeated call.

Reply:
done